tests/test_core/test_decorator_calls.py

```diff
@@ -25,33 +33,37 @@ def test_basic_decorator():
     list(conn.get_all_instances()).should.equal([])
 
 
-def test_context_manager():
+def test_context_manager(*args):
     conn = boto.connect_ec2('the_key', 'the_secret')
-    with assert_raises(EC2ResponseError):
-        conn.get_all_instances()
+    with mock.patch('boto.connection.AWSQueryConnection.make_request', return_value=MockHttpResponse()):
+        with assert_raises(EC2ResponseError):
+            conn.get_all_instances()
 
     with mock_ec2():
         conn = boto.connect_ec2('the_key', 'the_secret')
         list(conn.get_all_instances()).should.equal([])
 
-    with assert_raises(EC2ResponseError):
-        conn = boto.connect_ec2('the_key', 'the_secret')
-        conn.get_all_instances()
+    with mock.patch('boto.connection.AWSQueryConnection.make_request', return_value=MockHttpResponse()):
```

Comment:
The point of the `assert_raises(EC2ResponseError)` is to assert that the behavior goes back as expected once we are no longer patching. By patching `make_request`, it makes the test mostly useless. If we want to add 3.4, we'll need to figure out why this behavior doesn't work.